CrowdStrike/glide-core
# Switch to Rollup for components

## 🚀 Description

Esbuild's minifier mangles class names. This is a problem for `owSlotType` errors, whose error message includes the class name (first screenshot).

Unfortunately, Esbuild doesn't have an option to disable mangling class names. Terser does (`keep_classnames`) but doesn't have an Esbuild plugin. So I've swapped out Esbuild for Rollup, which Terser supports, as our TypeScript builder. 

We're still using Esbuild to bundle our stylesheets because I couldn't find a Rollup plugin that bundles CSS in a way that doesn't first involve importing it from a TypeScript file.

## 📋 Checklist



- I have read and followed the [Contributing Guidelines](https://github.com/crowdstrike/glide-core/blob/main/CONTRIBUTING.md).
- I have added tests to cover new or updated functionality.
- I have created or updated stories in Storybook to document the new functionality.
- I have included a changeset with this Pull Request if it adds/updates/removes functionality for consumers.
- I have scheduled a Design Review for these changes, if one is required.
- I have followed the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/) and/or met with the Accessibility Team to ensure this functionality is accessible.

## 🔬 How to Test

1. Check out the branch.
2. Run a production build.
3. Make sure class names aren't mangled.


If you're feeling ambitious, also check out [`ember-polaris-testing`](https://github.com/ynotdraw/polaris-ember-testing) and  run `pnpm link ../glide-core/packages/components` from that project's root. Make sure the components render. Then put an invalid element in Menu's default slot is show and check the error message.

## 📸 Images/Videos of Functionality

### Before

![image](https://github.com/CrowdStrike/glide-core/assets/114178960/79b0e980-105e-44aa-9d36-723e086c4263)

### After

<img width="1500" alt="image" src="https://github.com/CrowdStrike/glide-core/assets/114178960/ed840fe1-e139-4ff6-b1ec-bf5cb63c194a">
